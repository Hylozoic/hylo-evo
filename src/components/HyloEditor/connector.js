import { connect } from 'react-redux'
import { findMentions, clearMentions, getMentionResults } from './store'

export function mapStateToProps (state, props) {
  return {
    mentionResults: getMentionResults(state, props)
  }
}

export const mapDispatchToProps = {
  findMentions,
  clearMentions
}

export default connect(mapStateToProps, mapDispatchToProps)
