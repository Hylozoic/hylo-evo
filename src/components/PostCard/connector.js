import { connect } from 'react-redux'
import { push } from 'react-router-redux'

export function mapStateToProps (state, props) {
  return {}
}

export const mapDispatchToProps = {
  navigate: push
}

export default connect(mapStateToProps, mapDispatchToProps)
