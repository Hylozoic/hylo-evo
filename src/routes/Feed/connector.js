import { connect } from 'react-redux'
import { SAMPLE_FEED_ITEMS } from './sampleData'

function mapStateToProps (state) {
  return {
    feedItems: SAMPLE_FEED_ITEMS
  }
}

function mapDispatchToProps (dispatch, props) {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)
