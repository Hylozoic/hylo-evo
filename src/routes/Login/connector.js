import { connect } from 'react-redux'
import { checkLogin, login } from './actions'
export default connect(null, {checkLogin, login})
