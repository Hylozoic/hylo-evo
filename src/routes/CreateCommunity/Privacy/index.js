import component from './Privacy'
import connector from './Privacy.connector'
export default connector(component)
