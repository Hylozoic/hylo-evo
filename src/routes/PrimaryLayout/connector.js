import { connect } from 'react-redux'
import { SAMPLE_COMMUNITY } from 'routes/Feed/sampleData'
import { fetchCurrentUser } from './actions'

function mapStateToProps ({currentUser}) {
  return {
    community: SAMPLE_COMMUNITY,
    currentUser
  }
}

export default connect(mapStateToProps, {fetchCurrentUser})
