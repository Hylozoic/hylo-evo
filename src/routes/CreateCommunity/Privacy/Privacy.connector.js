import { connect } from 'react-redux'
import { push } from 'react-router-redux'
import { get } from 'lodash/fp'
import { addCommunityPrivacy } from '../CreateCommunity.store'

export function mapStateToProps (state, props) {
  return {
    communityPrivacy: get('privacy', state.CreateCommunity)
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToNextStep: () => dispatch(push('/create-community/review')),
    goToPreviousStep: () => dispatch(push('/create-community/domain')),
    addCommunityPrivacy: (privacy) => dispatch(addCommunityPrivacy(privacy)),
    goHome: () => dispatch(push('/'))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
