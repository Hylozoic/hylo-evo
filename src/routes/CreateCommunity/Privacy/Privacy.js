import React, { Component } from 'react'
import '../CreateCommunity.scss'
import ModalSidebar from 'components/ModalSidebar'
import { hyloNameWhiteBackground } from 'utils/assets'
import { bgImageStyle } from 'utils/index'
import ModalFooter from 'components/ModalFooter'
import Dropdown from 'components/Dropdown'
import Icon from 'components/Icon'
import { find } from 'lodash'
import { get } from 'lodash/fp'

export default class Privacy extends Component {
  constructor () {
    super()
    this.state = {
      // Set to 'private' by default
      selectedPrivacy: 1
    }
  }

  submit = () => {
    // TODO: Find out how this will actuall be saved in the db.
    // Currently set to be saved by label
    this.props.addCommunityPrivacy(privacyOptions[this.state.selectedPrivacy].label)
    this.props.goToNextStep()
  }

  selectPrivacy = (selectedPrivacy) => {
    this.setState({
      selectedPrivacy
    })
  }

  componentWillMount = () => {
    const { communityPrivacy } = this.props
    const privacyOption = find(privacyOptions, {label: communityPrivacy})
    this.setState({selectedPrivacy: get('id', privacyOption) || 1})
  }
  render () {
    return <div styleName='flex-wrapper'>
      <ModalSidebar
        onClick={this.props.goHome}
        theme={sidebarTheme}
        header="Great, let's get started"
        body="All good things start somewhere! Let's kick things off with a catchy name for your community."
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 3/4</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically privacy-label-row'>
          <span styleName='privacy-input-label'>SET THE PRIVACY LEVEL</span>
          <br />
          <span styleName='privacy-label'>This community is</span>
          <Dropdown styleName='privacy-dropdown'
            toggleChildren={<span>
              {privacyOptions[this.state.selectedPrivacy].label}
              <Icon name='ArrowDown' />
            </span>}
            items={privacyOptions.map(({ id, label }) => ({
              label,
              onClick: () => this.selectPrivacy(id)
            }))}
            alignRight />
        </div>
      </div>
      <ModalFooter
        submit={this.submit}
        previous={this.props.goToPreviousStep}
        continueText={'Continue'}
        />
    </div>
  }
}

const sidebarTheme = {
  sidebarHeader: 'sidebar-header-full-page',
  sidebarText: 'gray-text sidebar-text-full-page'
}

const privacyOptions = [
  {id: '0', label: 'public'},
  {id: '1', label: 'private'},
  {id: '2', label: 'unlisted'}
]
