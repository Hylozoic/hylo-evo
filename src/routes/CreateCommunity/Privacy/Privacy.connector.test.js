import { mapDispatchToProps, mapStateToProps } from './Privacy.connector'

const dispatch = jest.fn(x => x)
const props = {}
const dispatchProps = mapDispatchToProps(dispatch, props)

describe('Privacy', () => {
  it('should call addCommunityPrivacy from mapDispatchToProps', () => {
    const privacy = 'privacy'
    expect(dispatchProps.addCommunityPrivacy(privacy)).toMatchSnapshot()
  })

  it('should call goToPreviousStep from mapDispatchToProps', () => {
    expect(dispatchProps.goToPreviousStep()).toMatchSnapshot()
  })

  it('should call goToNextStep from mapDispatchToProps', () => {
    expect(dispatchProps.goToNextStep()).toMatchSnapshot()
  })

  it('should call goHome from mapDispatchToProps', () => {
    expect(dispatchProps.goHome()).toMatchSnapshot()
  })

  it('should have communityPrivacy in mapStateToProps', () => {
    const privacy = 'privacy'

    const state = {
      CreateCommunity: {
        privacy
      }
    }
    expect(mapStateToProps(state, props).communityPrivacy).toBe(privacy)
  })
})
