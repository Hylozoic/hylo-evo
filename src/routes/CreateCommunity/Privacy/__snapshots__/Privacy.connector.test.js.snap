// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Privacy should call addCommunityPrivacy from mapDispatchToProps 1`] = `
Object {
  "payload": "privacy",
  "type": "CreateCommunity/ADD_COMMUNITY_PRIVACY",
}
`;

exports[`Privacy should call goHome from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Privacy should call goToNextStep from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/create-community/review",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Privacy should call goToPreviousStep from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/create-community/domain",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
